src/components/UserProfile.re

```diff
@@ -227,79 +230,136 @@ module UserDetails = {
-          <br />
-          // NOTE: the number of loyalty tokens of a user currently will always show.
-          //       We are thinking of making this "private" only to the current logged in user. To enable this remove the `|| true` from the line below
-          {isAddressCurrentUser
-             ? <>
-                 <small>
-                   <p>
-                     {(
-                        "Claimed Loyalty Token Balance: "
-                        ++ {
-                          totalLoyaltyTokensOpt->Option.mapWithDefault(
-                            "Loading", claimedLoyaltyTokens => {
-                            claimedLoyaltyTokens->Web3Utils.fromWeiBNToEthPrecision(
-                              ~digits=6,
-                            )
-                          });
-                        }
-                        ++ " WLT"
-                      )
-                      ->restr}
-                   </p>
-                 </small>
-                 {switch (currentlyOwnedTokens) {
-                  | [||] => React.null
-                  | currentlyOwnedTokens =>
-                    currentlyOwnedTokens
-                    ->Array.map(id =>
-                        <ClaimLoyaltyTokenButtons
-                          id
-                          key=id
-                          refreshLoyaltyTokenBalance=updateFunction
-                        />
-                      )
-                    ->React.array
-                  }}
-                 <a href="/#ethturin-quadratic-voting"> "vote"->restr </a>
-               </>
-             : <small>
-                 <p>
-                   {(
-                      "Loyalty Token Balance Generated: "
-                      ++ {
-                        totalLoyaltyTokensAvailableAndClaimedOpt->Option.mapWithDefault(
-                          "Loading", ((totalLoyaltyTokens, _)) => {
-                          totalLoyaltyTokens->Web3Utils.fromWeiBNToEthPrecision(
-                            ~digits=5,
-                          )
-                        });
-                      }
-                      ++ " WLT"
-                    )
-                    ->restr}
-                 </p>
-               </small>}
+          {switch (nonUrlState) {
+           | UserVerificationScreen =>
+             <div className=Css.(style([position(`relative)]))>
+               <Rimble.Button.Text
+                 icononly=true
+                 icon="Close"
+                 color="moon-gray"
+                 position="absolute"
+                 top=0
+                 right=0
+                 m=1
+                 onClick={_ => clearNonUrlState()}
+               />
+               <React.Suspense fallback={<Rimble.Loader />}>
+                 <LazyThreeBoxUpdate.Lazy />
+               </React.Suspense>
+             </div>
+           | UpdateDepositScreen =>
+             <div className=Css.(style([position(`relative)]))>
+               <Rimble.Button.Text
```

Comment:
I could refactor this close button into a component actually. But not so important now.